packages/devtools_app/lib/src/ui/analytics.dart

```diff
@@ -154,17 +158,181 @@ class GtagExceptionDevTools extends GtagException {
+  } else {
+    log('HttpRequest api/getDevToolsEnabled failed status = ${resp.status}',
+        LogLevel.error);
+  }
+
+  _gaEnabled = enabled;
+
+  return enabled;
+}
+
+/// Set DevTools enabled (GA enabled) stored in ~/.devtools.
+void setEnabled([bool value = true]) async {
+  var resp = await HttpRequest.request(
+    '/api/setDevToolsEnabled?enabled=${value}',
+    method: 'POST',
+  );
+  if (resp.status == HttpStatus.ok) {
+    assert(resp.responseText == '$value');
+    _gaEnabled = value;
+  } else {
+    log(
+      'HttpRequest api/setDevToolsEnabled failed status = ${resp.status}, '
+      'responseText = ${resp.responseText}',
+      LogLevel.error,
+    );
+  }
+}
+
+/// Request DevTools surveyActionTaken property stored in ~\.devtools.
+Future<bool> get isSurveyActionTaken async {
+  bool surveyActionTaken = false;
+
+  var resp =
+      await HttpRequest.request('/api/getSurveyActionTaken', method: 'POST');
+  if (resp.status == HttpStatus.ok) {
+    surveyActionTaken = resp.responseText == 'true';
+  } else {
+    log('HttpRequest api/getSurveyActionTaken failed status = ${resp.status}',
+        LogLevel.error);
+  }
+  return surveyActionTaken;
+}
+
+/// Set DevTools surveyActionTaken property stored in ~\.devtools.
+void setSurveyActionTaken([bool value = true]) async {
+  var resp = await HttpRequest.request(
+    '/api/setSurveyActionTaken?surveyActionTaken=${value}',
+    method: 'POST',
+  );
+  if (resp.status == HttpStatus.ok) {
+    assert(resp.responseText == '$value');
+  } else {
+    log(
+      'HttpRequest api/setSurveyActionTaken failed status = ${resp.status}, '
+      'responseText = ${resp.responseText}',
+      LogLevel.error,
+    );
+  }
+}
+
+/// Request DevTools surveyShownCount property stored in ~\.devtools.
+Future<int> get surveyShownCount async {
+  int surveyShownCount;
+
+  var resp =
+      await HttpRequest.request('/api/getSurveyShownCount', method: 'POST');
+  if (resp.status == HttpStatus.ok) {
+    surveyShownCount = int.parse(resp.responseText);
+  } else {
+    log('HttpRequest api/getSurveyShownCount failed status = ${resp.status}',
+        LogLevel.error);
+  }
+  return surveyShownCount;
+}
+
+/// Increment DevTools surveyShownCount property stored in ~\.devtools.
+Future<int> get incrementSurveyShownCount async {
+  int surveyShownCount;
+
+  var resp = await HttpRequest.request('/api/incrementSurveyShownCount',
+      method: 'POST');
+  if (resp.status == HttpStatus.ok) {
+    surveyShownCount = int.parse(resp.responseText);
+  } else {
+    log('HttpRequest api/getSurveyShownCount failed status = ${resp.status}',
+        LogLevel.error);
+  }
+  return surveyShownCount;
+}
+
 void screen(
   String screenName, [
   int value = 0,
 ]) {
-  // Let the server know that we're logging a screen view event,
-  // then let GA know about the event.
-  HttpRequest.request(
-    '/api/logScreenView',
-    method: 'POST',
-    sendData: jsonEncode({'screen': screenName}),
-  );
```

Comment:
why did we remove this?

Reply:
It was to show how app to server would work.  This is not longer needed.